tests/Doctrine/Tests/DBAL/Schema/MySqlSchemaManagerTest.php

```diff
@@ -37,6 +38,79 @@ public function testCompositeForeignKeys()
         $this->assertEquals(array('column_1', 'column_2', 'column_3'), array_map('strtolower', $fkeys[0]->getForeignColumns()));
     }
 
+    public function testPortableTableColumnDefinition()
+    {
+        $eventManager = new EventManager();
+        $driverMock = $this->createMock('Doctrine\DBAL\Driver');
+
+        $platforms = [
+            'mysql' => new \Doctrine\DBAL\Platforms\MySqlPlatform(),
+            'mariadb10.2' => new \Doctrine\DBAL\Platforms\MariaDb102Platform()
+        ];
+
+        foreach ($platforms as $platformName => $platform) {
+
+            $conn = $this->getMockBuilder('Doctrine\DBAL\Connection')
+                ->setMethods(array('fetchAll'))
+                ->setConstructorArgs(array(array('platform' => $platform), $driverMock, new Configuration(), $eventManager))
+                ->getMock();
+            $this->manager = new MySqlSchemaManager($conn);
+
+            $conn->expects($this->once())->method('fetchAll')->will($this->returnValue($this->getColumnDefinition($platform)));
```

Comment:
IMO, there's no point mocking DB-specific features. A read DB should be used instead. Otherwise, it implies that we're in control of how different DBs behave which is not true.